src/offspot_demo/setup.py

```diff
@@ -1,2 +1,143 @@
+import shutil
+import subprocess
+import sys
+from time import sleep
+
+from offspot_demo.constants import (
+    DOCKER_COMPOSE_SYMLINK_PATH,
+    FQDN,
+    JINJA_ENV,
+    SRC_PATH,
+    STARTUP_DURATION,
+    SYSTEMD_UNIT_NAME,
+    SYSTEMD_UNIT_PATH,
+)
+
+
+def render_maint_caddyfile():
+    """Render the maintenance Caddyfile to customize local stuff"""
+    print("Rendering maintenance Caddyfile")
+    with open(SRC_PATH / "maint-compose/Caddyfile", "w") as fh:
+        fh.write(
+            JINJA_ENV.get_template("maint-compose/Caddyfile.tmpl").render(fqdn=FQDN)
+        )
+
+
+def render_maint_docker_compose():
+    """Render the maintenance docker-compose to customize local stuff"""
+    print("Rendering maintenance docker-compose")
+    with open(SRC_PATH / "maint-compose/docker-compose.yml", "w") as fh:
+        fh.write(
+            JINJA_ENV.get_template("maint-compose/docker-compose.tmpl.yml").render(
+                src_path=SRC_PATH
+            )
+        )
+
+
+def install_symlink():
+    """Install docker-compose symlink
+
+    Will be switched between maint and prod, but for now point to maint
+    """
+    print("Installing docker-compose symlink")
+    DOCKER_COMPOSE_SYMLINK_PATH.unlink(missing_ok=True)
+    DOCKER_COMPOSE_SYMLINK_PATH.symlink_to(
+        SRC_PATH / "maint-compose/docker-compose.yml"
+    )
+
+
+def run_command(
+    command: list[str], expected_return_code: int = 0
+) -> subprocess.CompletedProcess[str]:
+    """Run a shell command and check return code"""
+    status = subprocess.run(
+        [
+            "/usr/bin/env",
+            *command,
+        ],
+        text=True,
+        stderr=subprocess.STDOUT,
+        stdout=subprocess.PIPE,
+        check=False,
+    )
+    if status.returncode != expected_return_code:
+        print(f"Running command failed with code {status.returncode}")
+        print(f"Command was: {command}")
+        print("Stdout/Stderr is:")
+        print(status.stdout)
+        sys.exit(1)
+    return status
+
+
+def check_systemd_service(
+    expected_return_code: int = 0,
+    *,
+    check_running: bool = False,
+    check_enabled: bool = False,
+) -> subprocess.CompletedProcess[str]:
+    """Check status of the systemd unit
+
+    By default, check at least that the unit is loaded properly (i.e. parsing is ok)
+    If check_running is True, it also checks that the unit is running
+    If check_enabled is True, it also checks that the unit is enabled
+    """
+    status = run_command(
+        ["systemctl", "status", f"{SYSTEMD_UNIT_NAME}.service"],
```

Comment:
Dont you need `--no-pager` ?

Reply:
I will add it for safety, but nope I didn't need this